adding a JAVA client to test the home-recipes services

bhargavmehta01 opened the pull request.

bhargavmehta01 committed:
adding JAVA client for unary & server-side streamig service

bhargavmehta01 committed:
adding client implementation for client-side and bi-directional streamig services

bhargavmehta01 committed:
adding gradle task to run from terminal

bhargavmehta01 committed:
adding a README file for the JAVA client

kaushikchaubal commented:
I suggest to rename the branch from `app-client-java` to `test-client-java` to keep it consistent

kaushikchaubal commented on client/app-client-java/src/main/proto/google/api/annotations.proto:
```diff
@@ -0,0 +1,31 @@
+// Copyright (c) 2015, Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
```
Why is there a proto folder inside the client? Your client should pick up the protos from the defs folder. Please delete this folder and use the protos from the defs folder which  has all the definitions already

kaushikchaubal reviewed (changes requested):
Please update the two things  as mentioned in other comments:
1. Remove the duplicated proto folder and use the 'defs' folder which has all the proto definitions 
2. Rename the branch from app-client-java to test-client-java to keep it consistent

bhargavmehta01 committed:
renaming the project to suggested name

bhargavmehta01 committed:
removing duplicate proto defs and using the exisitng one from outside the JAVA project

bhargavmehta01 commented:
- Removed the duplicated proto defs folder and now the project depends on the existing proto defs outside the project
- Renamed to 'test-client-java' for the consistency purpose

kaushikchaubal commented:
thanks for your contribution! 💯
